packages/babel-parser/src/parser/expression.js

```diff
@@ -1214,6 +1214,26 @@ export default class ExpressionParser extends LValParser {
         return node;
       }
 
+      case tt.assign:
+        if (
+          this.state.value === "%=" &&
```

Comment:
Ｗe can create a new token type for `%=` like we did for `/=`, so we don't have to compare the string value here. Note that `this.state.value` is a slice from the input string so it is not a constant string, V8 will then fallback to StringEqual.